pkg/plugins/policies/meshtimeout/plugin/xds/configurer.go

```diff
@@ -148,6 +159,58 @@ func ConfigureRouteAction(
 	}
 }
 
+func ConfigureGatewayListener(
+	conf *api.Conf,
+	protocol core_mesh.Protocol,
+	listener *envoy_listener.Listener,
+) error {
+	if listener == nil {
+		return nil
+	}
+
+	httpTimeouts := func(hcm *envoy_hcm.HttpConnectionManager) error {
+		if hcm.CommonHttpProtocolOptions == nil {
+			hcm.CommonHttpProtocolOptions = &envoy_core.HttpProtocolOptions{}
+		}
+		hcm.CommonHttpProtocolOptions.IdleTimeout = toProtoDurationOrDefault(
+			pointer.Deref(conf).IdleTimeout,
+			policies_defaults.DefaultGatewayIdleTimeout,
+		)
+		hcm.RequestHeadersTimeout = toProtoDurationOrDefault(
+			pointer.Deref(conf.Http).RequestHeadersTimeout,
+			policies_defaults.DefaultGatewayRequestHeadersTimeout,
+		)
+		hcm.StreamIdleTimeout = toProtoDurationOrDefault(
+			pointer.Deref(conf.Http).StreamIdleTimeout,
+			policies_defaults.DefaultGatewayStreamIdleTimeout,
+		)
+		if httpConf := pointer.Deref(conf.Http); httpConf.RequestTimeout != nil {
+			hcm.RequestTimeout = util_proto.Duration(httpConf.RequestTimeout.Duration)
+		}
+		return nil
```

Comment:
I'm kinda thinking now should we have logic for applying default timeouts in two places, here and: https://github.com/kumahq/kuma/blob/8cc520ea4c14e20b386702fcf41056c66d44f33b/pkg/plugins/runtime/gateway/filter_chain_generator.go#L205-L209

Reply:
One thought occurred to me: Previously, users were unable to configure these timeouts, and if the default configuration did not suit them, they had to use a ProxyTemplate to modify it. By establishing defaults in the plugin code, we risk overriding the user's preferences. I am thinking about tweaking the logic to only update values if the current ones are the default settings.

Reply:
I've tested it locally it seems ProxyTemplate is applied later for Gateway. It's kinda interesting because it's applied even after MeshProxyPatch